add monarch.torchrun module; replicate torchrun using monarch as a proc manager

colin2328 opened the pull request:
Differential Revision: D86152013

python -m monarch.actor.torchrun train.py
NCCL version 2.27.5+cuda12.9
[Rank 0] Step 0 loss=-3.3797261714935303
[Rank 1] Step 0 loss=-2.670731544494629
[Rank 0] Step 1 loss=0.08844298124313354
[Rank 1] Step 1 loss=-1.594184398651123
[Rank 0] Step 2 loss=-2.7455830574035645
[Rank 1] Step 2 loss=-0.4763872027397156
[Rank 0] Step 3 loss=-2.7619757652282715
[Rank 1] Step 3 loss=-2.6958324909210205
[Rank 0] Step 4 loss=-1.0318412780761719
[Rank 1] Step 4 loss=-3.984520673751831

ezyang reviewed (approved):
Thanks! Is there a design goal to match the CLI interface of torchrun exactly?

colin2328 committed:
add monarch.torchrun module; replicate torchrun using monarch as a proc manager (#1750)

Summary:

adds a new monarch.actor.torchrun module. with a torchrun interface.
https://fb.workplace.com/groups/996292674996363/permalink/1318076819484612/

torchx app def is next diff D86155019

Reviewed By: ezyang

Differential Revision: D86152013
